### PR TITLE
v1.16 Backports 2024-10-07

### PR DESCRIPTION
**Pull request by giorio94**
* [x] #34090 (@dlapcevic) :warning: resolved conflicts
    * :information_source: Hit minor conflict due to different surrounding context, accepted combination.
 * [x] #35107 (@aanm)
 * [ ] #35033 (@WeeNews)
 * [ ] #34808 (@cfsnyder)
   * :information_source: Added missing `ptr` package import in `ingestion/gateway_test.go`.
 * [ ] #35068 (@ayuspin)
 * [ ] #35109 (@jrajahalme) :warning: resolved conflicts
     * :information_source: Hit minor conflict due to different surrounding context, accepted combination. Additionally updated the `TestMapState_AccumulateMapChangesOnVisibilityKeys` test, which has been removed in the meanwhile on main, applying the changes originally part of cilium/cilium@137abf7b4b6a (from #35109).
 * [x] #35193 (@aanm)
 * [ ] #35070 (@jrife)
 * [ ] #35126 (@jrife)
 * [x] #35177 (@marseel)
 * [x] #35208 (@aanm)
 * [ ] #35186 (@kreeuwijk) :warning: resolved conflicts
    * :information_source: Hit conflict as c8ee53c0014c is not yet present in v1.16. Adapted the patch accordingly.
 * [x] #35166 (@rastislavs)
    * :information_source: Switched to the experimental maps package (and dropped slices.Collect) in `reconcilerv2/service_test.go`, given that v1.16 is still using go v1.22.
 * [x] #35262 (@aanm)
 * [x] #35264 (@aanm)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 34090 35107 35033 34808 35068 35109 35193 35070 35126 35177 35208 35186 35166 35262 35264
```

### CODE REVIEW

**Review by rastislavs — APPROVED** [review]
LGTM for my commit, thanks!

**Comment by giorio94** [comment]
/test-backport-1.16

**Review by aanm — APPROVED** [review]

**Review by marseel — APPROVED** [review]

**Review by dlapcevic — APPROVED** [review]

**Comment by giorio94** [comment]
/test-backport-1.16

**Review by julianwiedmann — APPROVED** [review]

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 1]
```diff
@@ -432,6 +432,20 @@ jobs:
             ciliumendpointslice: 'true'
             skip-upgrade: 'true'
 
+          - name: '23'
```
Comment (brlbil) [inline .github/workflows/tests-e2e-upgrade.yaml]: This should be `27` and `max-parallel: 26` should also be changed to `max-parallel: 27`

**Review by brlbil — CHANGES_REQUESTED** [review]

File: .github/workflows/tests-e2e-upgrade.yaml (hunk 1 above)
Comment (giorio94) [reply .github/workflows/tests-e2e-upgrade.yaml]: Good catch, thanks! Should be good now.

**Comment by giorio94** [comment]
/test-backport-1.16

**Review by brlbil — APPROVED** [review]
Thanks

**Pull request merged**